src/library/scala/util/Either.scala

```diff
@@ -457,6 +457,35 @@ object Either {
   def cond[A, B](test: Boolean, right: => B, left: => A): Either[A, B] =
     if (test) Right(right) else Left(left)
 
+  /**
+   * Creates `FromLeftPartiallyApplied[B]` that given an argument of type `A`
+   * produces a `Left` typed as `Either[A,B]`
+   * {{
+   *   val l = Either.left[Int]("boo")
+   *   //l: Either[String, Int] = Left("boo")
+   * }}
+   */
+  def left[B]: FromLeftPartiallyApplied[B] = new FromLeftPartiallyApplied[B]
+
+  private[util] final class FromLeftPartiallyApplied[B](private val dummy: Unit) extends AnyVal {
```

Comment:
@why `private[util]` here (and 14 lines down), why not fully `private`? (Guess: is it because otherwise you get a warning about the type escaping...?)

Reply:
are there any pros/cons to using `Unit` vs `Boolean` in this pattern?

Reply:
I'm pretty sure I've tried to make it fully `private`, but compiler wasn't happy. I will double check it

Reply:
As for Unit, i guess it's just less to type, no need to give it default value. In any case it's just a trick for `AnyVal`. 
Maybe someone has a better idea here, as I don't see much difference otherwise

Reply:
I think I saw somewhere that using `Unit` didn't work until Scala 2.12.3. that would explain it

Reply:
as expected
```
[error] /home/jozic/projects/github/scala/src/library/scala/util/Either.scala:468: private class FromLeftPartiallyApplied escapes its defining scope as part of type scala.util.Either.FromLeftPartiallyApplied[B]
[error]   def left[B]: FromLeftPartiallyApplied[B] = new FromLeftPartiallyApplied[B]
[error]                ^
[error] one error found
[error] (library/compile:compileIncremental) Compilation failed
```